File: app/src/main/java/com/easysteps/activity/OtpActivity.java
package com.easysteps.activity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.easysteps.R;
import com.easysteps.helper.Utils;
import com.easysteps.model.loginsignup.OtpRes;
import com.easysteps.retrofit.AsyncHttpRequest;
import com.easysteps.retrofit.AsyncResponseHandler;
import com.easysteps.retrofit.RequestParamsUtils;
import com.easysteps.retrofit.URLs;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import okhttp3.Call;
import okhttp3.MultipartBody;

public class OtpActivity extends AppCompatActivity {

    String email;
    private EditText edt_otp;
    private EditText edt_new_password;
    private EditText edt_confirm_password;
    private TextView txt_send;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_otp);

        Intent intent = getIntent();
        email = intent.getStringExtra("email");


        edt_otp = (EditText) findViewById(R.id.edt_otp);
        edt_new_password = (EditText) findViewById(R.id.edt_new_password);
        edt_confirm_password = (EditText) findViewById(R.id.edt_confirm_password);
        txt_send = (TextView) findViewById(R.id.txt_send);



        txt_send.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (edt_new_password.getText().toString().trim().equals(edt_confirm_password.getText().toString().trim())){
                RestPassword();
                }else {
                    Toast.makeText(OtpActivity.this, "Please Enter Valid Password!", Toast.LENGTH_SHORT).show();
                }
            }
        });

    }

    private void RestPassword() {
        try {
            Utils.ShowProgressDialog(this);
            MultipartBody.Builder body = RequestParamsUtils.newRequestMultipartBody(this);
            body.setType(MultipartBody.FORM);
            body.addFormDataPart(RequestParamsUtils.userOtp, "121212");
            body.addFormDataPart(RequestParamsUtils.userNewPass, edt_confirm_password.getText().toString().trim());
            body.addFormDataPart(RequestParamsUtils.userEmail, email);

            Call call = AsyncHttpRequest.newRequestPost(this, body.build(), URLs.RESET_PASSWORD());
            call.enqueue(new RestPasswordHandler(this));
        }catch (Exception e){
            Log.e("TAG", "ForgotPassword: "+e.getMessage() );
        }

    }

    private class RestPasswordHandler extends AsyncResponseHandler {
        RestPasswordHandler(Activity activity) {
            super(activity);
        }

        @Override
        public void onStart() {

        }

        @Override
        public void onSuccess(String responce) {
            new Handler(Looper.getMainLooper()).post(() -> {

                Utils.HideProgressDialog(OtpActivity.this);

                if (responce != null && responce.length() > 0) {
                    OtpRes otpRes = new Gson().fromJson(responce, new TypeToken<OtpRes>() {
                    }.getType());
                    if (otpRes.getStatus()==1){
                        startActivity(new Intent(OtpActivity.this,LoginActivity.class));
                        finish();
                    }else {
//                        Utils.MyShortToast(OtpActivity.this,otpRes.getMessage());
                    }
                }
            });
        }

        @Override
        public void onFinish() {
            try {
                Utils.HideProgressDialog(OtpActivity.this);
            } catch (Exception e) {
                e.printStackTrace();
            }
        }

        @Override
        public void onFailure(Throwable e, String content) {
            Log.e("TAG", "onFailure: "+ e.getMessage() );
            Utils.HideProgressDialog(OtpActivity.this);
        }
    }

}

File: app/src/main/java/com/easysteps/activity/OtpActivity.kt
package com.easysteps.activity

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.activity.viewModels
import com.easysteps.R
import com.easysteps.base.BaseActivity
import com.easysteps.databinding.ActivityOtpBinding
import com.easysteps.helper.EXTRA_EMAIL
import com.easysteps.helper.RequestParamsUtils
import com.easysteps.helper.Utils
import com.easysteps.viewModel.VerifyOtpViewModel

/**
 * Created by NIKUNJ on 03-05-2022.
 */

class OtpActivity : BaseActivity<ActivityOtpBinding>(R.layout.activity_otp), View.OnClickListener {

    var email: String? = null
    val viewModel: VerifyOtpViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        initView()
        setUpObserver()
        setOnClickListener()
    }

    private fun setOnClickListener() {
        binding.tvSend.setOnClickListener(this)
    }

    private fun initView() {
        if (intent.hasExtra(EXTRA_EMAIL)) {
            email = intent.getStringExtra(EXTRA_EMAIL)
        }
    }

    private fun setUpObserver() {
        viewModel.apiErrors.observe(this) { handleError(it) }

        viewModel.appLoader.observe(this) { updateLoaderUI(it) }

        viewModel.dataError.observe(this) { handleError(it) }

        viewModel.verifyOtpData.observe(this) {
            if (it.status == 1) {
                finishAffinity()
                startActivity(Intent(this@OtpActivity, LoginActivity::class.java))
            } else Utils.MyShortSnackbar(binding.llLayout, it.message)
        }
    }

    override fun onClick(p0: View?) {
        when (p0?.id) {
            R.id.tvSend -> {
                if (isValid()) {
                    callVerifyOtpApi()
                }
            }
        }
    }

    private fun callVerifyOtpApi() {
        val map = HashMap<String, Any>()

        map[RequestParamsUtils.userOtp] = "121212"
        map[RequestParamsUtils.userNewPass] = binding.etConfirmPassword.text.toString().trim()
        map[RequestParamsUtils.userEmail] = email.toString()

        viewModel.callApi(map)
    }

    private fun isValid(): Boolean {
        when {
            binding.etOtp.text.toString().trim().isEmpty() -> {
                Utils.MyShortSnackbar(binding.llLayout, getString(R.string.text_empty_email))
                return false
            }
            binding.etNewPassword.text.toString().trim().isEmpty() -> {
                Utils.MyShortSnackbar(binding.llLayout, getString(R.string.text_empty_mobile))
                return false
            }
            binding.etConfirmPassword.text.toString().trim().isEmpty() -> {
                Utils.MyShortSnackbar(binding.llLayout, getString(R.string.text_empty_password))
                return false
            }
            binding.etConfirmPassword.text.toString().trim() != binding.etNewPassword.text.toString().trim() -> {
                binding.etConfirmPassword.setText("")
                binding.etNewPassword.setText("")
                return false
            }
            else -> return true
        }
    }

}